### PR TITLE
55: Use typescript for tests

### PR DESCRIPTION
**Pull request by marcusedwardhaslam**
https://github.com/sgb-io/codehawk-cli/issues/55

- Use ts-jest
- Remove specific test directory and use inline spec or test files
- Change existing .js test files to .ts

### CODE REVIEW

File: src/codehawk.test.ts [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-} = require('../build/codehawk')
-const { formatResultsAsTable } = require('../build/cli-util')
-const { JsxEmit } = require('typescript')
+import * as fs from 'fs'
+import { analyzeProject, calculateComplexity, generateBadge } from './codehawk'
+import { formatResultsAsTable } from './cli-util'
 
 const cwd = process.cwd()
-const outputMatchesResult = (projectPath) => {
+function outputMatchesResult(projectPath: string): void {
```
Comment (sgb-io) [inline src/codehawk.test.ts]: think `const` should be ok? would match the existing style

File: .eslintrc.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "plugins": ["@typescript-eslint", "prettier"],
   "extends": ["standard-with-typescript", "prettier", "prettier/@typescript-eslint", "eslint-config-prettier"],
   "parserOptions": {
-    "project": "./tsconfig.json",
+    "project": "./tsconfig.eslint.json",
```
Comment (sgb-io) [inline .eslintrc.json:7]: I think this means the new file is the only one used during the build? Therefore the options are incorrect and the build might be off

File: src/options.test.ts [hunk 3]
```diff
@@ -14,11 +16,14 @@ describe('when attempting to get the configuration for codehawk', () => {
         good: 'yes',
       }
 
-      require('fs').__codehawkConfigFound(true, mockCodehawkJson)
+      ;(fs.existsSync as jest.Mock).mockReturnValue(true)
```
Comment (sgb-io) [inline src/options.test.ts]: semicolons wat?

Comment (marcusedwardhaslam) [reply src/options.test.ts]: This is an automated prettier fix :(

I guess the interpreter looks at the object above and attempts to "call" it with the `fs.existsSync`

Comment (sgb-io) [reply src/options.test.ts]: kinda crazy! but fair enough

File: .eslintrc.json (hunk 2 above)
Comment (sgb-io) [reply .eslintrc.json:7]: Will double check this soon.. i just want be totally sure that we end up with the right output

File: src/options.test.ts (hunk 3 above)
Comment (marcusedwardhaslam) [reply src/options.test.ts]: Fixed!

File: .eslintrc.json (hunk 2 above)
Comment (sgb-io) [reply .eslintrc.json:7]: I was talking rubbish - this only affects what eslint sees.

**Comment by sgb-io** [comment]
Much simpler than I was making it, thank you @marcusedwardhaslam! Resolves #55

**Pull request merged**